### DISCUSSION TITLE
deprecating the current usage, and make it follow official CSS Module spec

### DISCUSSION THREAD

**Discussion by trusktr** [indooorsman/esbuild-css-modules-plugin#66] [Ideas]
Here's how the standardized CSS Module spec works:

https://web.dev/css-module-scripts

The import does not return CSS class names, but a CSS style sheet.

It could be possible to provide a utility that gets the class names from the sheet, and even modified the class names in the CSS OM, and it could also be possible to do at build time. With this idea, the default behavior would be as the CSS Module spec, but then some options could be specified to opt into the non-standard behavior.

The standard behavior would be more useful for vanilla custom elements and aligned with future direction of web, but the opt-in CSS name handling could be more useful for systems like React, Preact, Solid.js, etc.

**Comment by indooorsman** [comment]
That `module` I think is totally different with the one in [css modules](https://github.com/css-modules/css-modules).
As mentioned in https://web.dev/css-module-scripts I think it's just like a normal export of `esm` without any changes to class names.
However it's a cool idea, I will have a try, thanks @trusktr  !

**Comment by indooorsman** [comment]
@trusktr after some investigation I found that import assertion has been deprecated and there's another propsal using `with` instead of `assert` https://github.com/tc39/proposal-import-attributes and `esbuild` hasn't implemented this feature.
